### PR TITLE
docs: release guide

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

close https://github.com/chaos-mesh/chaos-mesh/issues/2760

### What's changed and how it works?

Appending new document `release.md` about how to publish new release

[rendered markdown](https://github.com/STRRL/chaos-mesh/blob/release-guideline/RELEASE.md)
### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by STRRL** [comment]
PTAL

/cc @chaos-mesh/committers @chaos-mesh/maintainers

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Release Guide
+
+This document introduces how to publish a new release of Chaos Mesh.
+
+## How to publish a new release
+
+Here are several steps to publish a new release, and several of them are optional for bugfix/patch relese.
```
Comment (fewdan) [inline RELEASE.md:7]: bugfix/patch release.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+### Step 1 Draft Release Notes
+
+This step is required for all the release(major, minor, bugfix/patch).
```
Comment (fewdan) [inline RELEASE.md:11]: all the releases

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+### Step 2 Checkout New Branch
+
+This step is only required for major or minor version release. You should skip this step if you are going to release a bugfix/patch version.
+
+Checkout a new branch with the name of `release-x.y` from `master` where `x.y` is the semantic version number of the release.
+
+### Step 3 Update Versions in Helm Charts and install.sh
+
+This step is required for all the release(major, minor, bugfix/patch).
```
Comment (fewdan) [inline RELEASE.md:28]: ditto

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  - After `2.1.0`, there is only one place need to change: `images.tag` in `helm/chaos-mesh/values.yaml`.
+  - Before `2.1.0`, you should update the image tag for **each images**
+- `version` in help messages of `install.sh`, about line 46.
+- Execute `make check` for update the versions in generated files.
+
+Then you could make a PR with above changes into `release-x.y` branch, here is an example: https://github.com/chaos-mesh/chaos-mesh/pull/2631
+
+### Step 4 Create Release on GitHub
+
+This step is required for all the release(major, minor, bugfix/patch).
```
Comment (fewdan) [inline RELEASE.md:43]: ditto

File: RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+Draft a new Release on GitHub Release: https://github.com/chaos-mesh/chaos-mesh/releases/new with the Release Notes from Step 1, and choose the `release-x.y` branch for release/tag `vx.y.z`.
+
+Please note that here requires prefix `v` on the tag.
+
+### Step 5 Build and Publish Docker Images
+
+This step is required for all the release(major, minor, bugfix/patch).
```
Comment (fewdan) [inline RELEASE.md:51]: ditto

File: RELEASE.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+After 2.1.0, a GitHub Action would automatically run after GitHub Release is published: https://github.com/chaos-mesh/chaos-mesh/actions?query=event%3Arelease
+
+Before 2.1.0, you should manually trigger a Jenkins Pipeline with several parameters: `tag` and `branch`. https://ci.pingcap.net/view/chaos-mesh/job/release_chaos_mesh/
+
+### Step 6 Upload crd.yaml and install.sh to CDN
+
+This step is required for all the release(major, minor, bugfix/patch).
```
Comment (fewdan) [inline RELEASE.md:59]: ditto

File: RELEASE.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+A GitHub Action would automatically run after GitHub Release is published: https://github.com/chaos-mesh/chaos-mesh/actions/workflows/upload_release_files.yml
+
+### Step 7 Build Helm Charts
+
+This step is required for all the release(major, minor, bugfix/patch).
```
Comment (fewdan) [inline RELEASE.md:65]: ditto

File: RELEASE.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+### Step 3 Update Versions in Helm Charts and install.sh
+
+This step is required for all the release(major, minor, bugfix/patch).
+
+You should make a new PR for updating the version in helm charts and `install.sh`:
+
+- `version` and `appVersion` in `helm/chaos-mesh/Chart.yaml`. Please notice that do NOT use prefix `v` in here.
+- Docker image tags in helm charts:
+  - After `2.1.0`, there is only one place need to change: `images.tag` in `helm/chaos-mesh/values.yaml`.
+  - Before `2.1.0`, you should update the image tag for **each images**
```
Comment (fewdan) [inline RELEASE.md:35]: maybe 'each image' ?

File: RELEASE.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+- `version` in help messages of `install.sh`, about line 46.
+- Execute `make check` for update the versions in generated files.
```
Comment (fewdan) [inline RELEASE.md:37]: for "updating" the versions

File: RELEASE.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+A GitHub Action would automatically run after GitHub Release is published: https://github.com/chaos-mesh/chaos-mesh/actions/workflows/upload_release_files.yml
+
+### Step 7 Build Helm Charts
+
+This step is required for all the release(major, minor, bugfix/patch).
+
+- Pull the latest code from `release-x.y` branch
+- `git tag chart-vx.y.z`
+- `git push upstream chart-vx.y.z`(`upstream` is the a remote points to `github.com/chaos-mesh/chaos-mesh`)
```
Comment (fewdan) [inline RELEASE.md:69]: “   `upstream` is the a remote points   ”   maybe " `upstream` is the remote point"  ?

File: RELEASE.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+- A GitHub Action would automatically run: https://github.com/chaos-mesh/chaos-mesh/actions/workflows/release_helm_chart.yml. And new helm artifact will be published to https://github.com/chaos-mesh/charts/tree/gh-pages.
+
+### Step 8 Update TiChi Bot Configuration
+
+This step is only required for major or minor version release. You should skip this step if you are going to release a bugfix/patch version.
+
+If new branches are created, you should update the TiChi Bot configuration:
+
+- Make a PR for setting up new branch: https://github.com/ti-community-infra/configs/blob/992ff03161a42c0b517e4b4239adbf5f94e96a50/prow/config/config.yaml#L1105. Configuration for new branch could be found in existing settings for other branches.
```
Comment (fewdan) [inline RELEASE.md:78]: setting up a new branch

File: RELEASE.md (hunk 10 above)
Comment (STRRL) [reply RELEASE.md]: I update it as "`upstream` is the remote repo `github.com/chaos-mesh/chaos-mesh`", how do you think about it? :D

**Comment by STRRL** [comment]
Updated! PTAL @fewdan

**Comment by iguoyr** [comment]
@STRRL Hi, any updates here? :)

**Comment by STRRL** [comment]
> @STRRL Hi, any updates here? :)

I would update this PR after we applied the CHANGELOG. There are something unresolved like "how to convert CHANGELOG to Release Notes", I  think I could update it after we release v2.1.3. :D

**Comment by STRRL** [comment]
> > @STRRL Hi, any updates here? :)
> 
> I should update this PR after we applied the CHANGELOG. There are something unresolved like "how to convert CHANGELOG to Release Notes", I think I could update it after we release v2.1.3. :D

I have updated the parts about CHANGELOG, PTAL! 

/cc @iguoyr @milasuperstar

File: RELEASE.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+### Step 4 Update CHANGELOG.md on target branch
+
+This step is required for all the releases(major, minor, bugfix/patch).
+
+You should update the `CHANGELOG.md` on the target branch:
+
+- rename `[Unreleased]` to `[vx.y.z] - YYYY-MM-DD`, and slim empty changes types with only `-Nothing`.
+- create new `[Unreleased]` at the top of `CHANGELOG.md`, with `- Nothing` placeholder in each type of changes.
+
+For more detail, see RFC [keep-a-changelog#](https://github.com/chaos-mesh/rfcs/blob/main/text/2022-01-17-keep-a-changelog.md#changelogmd-in-release--branches)
```
Comment (iguoyr) [inline RELEASE.md:54]:
```suggestion
For more detail, see RFC [keep-a-changelog](https://github.com/chaos-mesh/rfcs/blob/main/text/2022-01-17-keep-a-changelog.md#changelogmd-in-release--branches)
```

File: RELEASE.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+Draft a new Release on GitHub Release: https://github.com/chaos-mesh/chaos-mesh/releases/new with the Release Notes from Step 1, and choose the `release-x.y` branch for release/tag `vx.y.z`.
+
+Please note that here the prefix `v` on the tag is required.
+
+### Step 6 Build and Publish Docker Images
+
+This step is required for all the releases(major, minor, bugfix/patch).
+
+After 2.1.0, a GitHub Action would automatically run after GitHub Release is published: https://github.com/chaos-mesh/chaos-mesh/actions?query=event%3Arelease
```
Comment (iguoyr) [inline RELEASE.md:68]:
```suggestion
After `2.1.0`, a GitHub Action would automatically run after GitHub Release is published: https://github.com/chaos-mesh/chaos-mesh/actions?query=event%3Arelease
```

File: RELEASE.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+Before 2.1.0, you should manually trigger a Jenkins Pipeline with several parameters: `tag` and `branch`. https://ci.pingcap.net/view/chaos-mesh/job/release_chaos_mesh/
```
Comment (iguoyr) [inline RELEASE.md:70]:
```suggestion
Before `2.1.0`, you should manually trigger a Jenkins Pipeline with several parameters: `tag` and `branch`. https://ci.pingcap.net/view/chaos-mesh/job/release_chaos_mesh/
```

File: RELEASE.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+This step is required for all the releases(major, minor, bugfix/patch).
+
+You should make a new PR for updating the version in helm charts and `install.sh`:
+
+- `version` and `appVersion` in `helm/chaos-mesh/Chart.yaml`. Please notice that do NOT use prefix `v` in here.
+- Docker image tags in helm charts:
+  - After `2.1.0`, there is only one place need to change: `images.tag` in `helm/chaos-mesh/values.yaml`.
+  - Before `2.1.0`, you should update the image tag for **each image**
+- `version` in help messages of `install.sh`, about line 46.
```
Comment (iguoyr) [inline RELEASE.md:40]: > about line 46

After https://github.com/chaos-mesh/chaos-mesh/pull/2797, it should be line 45 now 😢  maybe it's not a good idea to clarify the line number.

File: RELEASE.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+
+## How to publish a new release
+
+Here are several steps to publish a new release, and several of them are optional for bugfix/patch release.
+
+### Step 1 Draft Release Notes
+
+This step is required for all the releases(major, minor, bugfix/patch).
+
+As we maintain a CHANGELOG.md on each active branch, we could use it to draft release notes. Changelog is used for developers and Release Notes is used for users, so there are some little different between them.
```
Comment (iguoyr) [inline RELEASE.md:13]:
```suggestion
As we maintain a `CHANGELOG.md` on each active branch, we could use it to draft release notes. Changelog is used for developers and Release Notes is used for users, so there are some little different between them.
```

**Review by iguoyr — COMMENTED** [review]
Rest LGTM

File: RELEASE.md (hunk 15 above)
Comment (STRRL) [reply RELEASE.md]: I change the way to describe it , PTAL :D

File: RELEASE.md [hunk 17]
```diff
@@ -23,10 +23,26 @@ As we maintain a `CHANGELOG.md` on each active branch, we could use it to draft
...
-This step is only required for major or minor version release. You should skip this step if you are going to release a bugfix/patch version.
+**This step is only required for major or minor version release.** You should skip this step if you are going to release a bugfix/patch version.
 
 Checkout a new branch with the name of `release-x.y` from `master` where `x.y` is the semantic version number of the release.
 
+Change the content in `env-images.yaml`, update the branch name from `master` to `release-x.y`.
+
+For example, if we are going to release `10.11.0`, you should:
+
+- checkout new branch `release-10.11.0` from `master`
```
Comment (iguoyr) [inline RELEASE.md:34]:
```suggestion
- checkout new branch `release-10.11` from `master`
```

File: RELEASE.md [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+You should make a new PR for updating the version in helm charts and `install.sh`:
+
+- `version` and `appVersion` in `helm/chaos-mesh/Chart.yaml`. Please notice that do NOT use prefix `v` in here.
+- Docker image tags in helm charts:
+  - After `2.1.0`, there is only one place need to change: `images.tag` in `helm/chaos-mesh/values.yaml`.
+  - Before `2.1.0`, you should update the image tag for **each image**
+- `version` in help messages of `install.sh`, like `Version of chaos-mesh, default value: <replace-with-version>`.
+- Execute `make check` for updating the versions in generated files.
+
+Then you could make a PR with above changes into `release-x.y` branch, here is an example: https://github.com/chaos-mesh/chaos-mesh/pull/2631
```
Comment (iguoyr) [inline RELEASE.md:59]: I think the action 'make a PR' should be put in 'Step 4'.

**Comment by STRRL** [comment]
it requires following https://github.com/chaos-mesh/chaos-mesh/issues/3172

**Comment by STRRL** [comment]
I would create the Release Guide on Wiki.

File: RELEASE.md (hunk 18 above)
Comment (STRRL) [reply RELEASE.md:59]: after https://github.com/chaos-mesh/chaos-mesh/pull/3293 merged, we do not need to modify that manually anymore. :D

**Pull request closed without merging**